grails-app/conf/SecurityFilters.groovy

```diff
@@ -117,6 +118,12 @@ class SecurityFilters {
                     redirect(controller: 'auth', action: 'login')
                     return false
                 }
+                // Check if a user is logged to disabled location (location.active or organization.active is false) - if so, redirect to location chooser
+                Location currentLocation = Location.get(session?.warehouse?.id)
```

Comment:
Let's get the current location from the thread local (which is set on line 49 of SecurityFilters)
```
Location currentLocation = AuthService.currentLocation.get()
```

If there are cases where this returns null (i.e. this line of code is invoked before line 49) then we could so something like this.
```
Location currentLocation = AuthService.currentLocation.get()?:Location.get(session?.warehouse?.id)
```

Reply:
I want to avoid doing multiple queries for the same data in the same request.